### PR TITLE
Fix rebin of phasequads in muon GUI

### LINKED ISSUE #32519: Muon phasequad bugs
Reported by James

There are a few bugs:
1. Rebinned phasequads have the wrong amplitude. It is better to rebin the raw data
2. Does not select the phasequad correctly in a sim fit

**Describe the bug**

**To Reproduce**
Set binning 4
Load MUSR run 62260
Create a phase table e.g. top, bottom
Create a Phase Quad from it
De-select the plain groups in the Grouping tab leaving only the Phase Quad for fitting
Go to Fitting. Note the phase quad amplitude in the plot pane is about 4 times what it should be
Select ExpDecayOsc. Set Simultaneous over Run. Make A, Frequency, Lambda be Global
Un-select “Fit to Raw Data” below
Try a fit. Get “Unknown property search object InputWorkspace_2”
Go back to Phase Table. Create a 2nd phase table e.g. bkwd, fwd
Try selecting it. Crash

**Expected behavior**

**Screenshots**

**Platform/Version (please complete the following information):**
 - OS: [e.g. Windows, RHEL 7, Ubuntu, macOS]
 - OS Version:
 - Mantid Version [e.g. 6.0.0]

**Additional context**

### PR DESCRIPTION
**Pull request by AnthonyLim23**
**Description of work.**
The phasequads in the muon GUI had 2 problems. The first was that it was possible to cause a hard crash when using binned data. The second was that the binned data was not normalised.

Reported by James

**To test:**

Set binning to 4 (or use the log rebin args)
Load MUSR run 62260
Go to the phase tab
Create a phase table e.g. top, bottom (press the button and enter a name in the pop up)
Create a Phase Quad from it by pressing the `+` button and give it a name
Go to the grouping tab
De-select the groups (top table) 
Go to Fitting tab
Tick the Simultaneous box
Add an ExpDecayOsc. . Make A, Frequency, Lambda be Global
Un-select “Fit to Raw Data” below
Try a fit. This is where it used to throw an error.

Go to the grouping tab
Select one of the groups (which does not matter)
Go back to the home tab
Try different rebin parameters in the rebin box (it takes the same args as the alg)
Make sure that the plots of the phasequad are roughly the same amplitude as the group you have selected (about 0.3)

Fixes #32519. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Muon/GUI/Common/ADSHandler/ADS_calls.py [hunk 1]
```diff
@@ -26,3 +27,11 @@ def check_if_workspace_exist(name):
 
 def add_ws_to_ads(name, workspace):
     AnalysisDataService.addOrReplace(name, workspace)
+
+
```
Comment (sf1919) [inline scripts/Muon/GUI/Common/ADSHandler/ADS_calls.py:31]: Is this one of those fun Github errors and there isn't actually two empty lines here?

Comment (AnthonyLim23) [reply scripts/Muon/GUI/Common/ADSHandler/ADS_calls.py:31]: there are 2 empty lines for pep8 (2 lines between functions)

**Review by sf1919 — CHANGES_REQUESTED** [review]
I have tested the PR and everything works as expected. There is certainly no longer a crash.

Reviewing the code its good to see that tests have been updated to reflect the new code. Most of the new functions are just calling algorithms that should already be tested elsewhere. However I could not see a test specifically for ```_average_by_bin_widths```. I see that the tests check whether or not this has been called but not the function itself. Could you add one please?

**Review by sf1919 — APPROVED** [review]
On closer inspection there is a test for the new function so changes do not need to be made. 

I have re-reviewed the code and am happy to approve it.

**Pull request merged**